src/rest_vol_attr.c

```diff
@@ -1763,6 +1763,19 @@ RV_attr_specific(void *obj, const H5VL_loc_params_t *loc_params, H5VL_attr_speci
                      */
                     switch (parent_obj_type) {
                         case H5I_FILE:
+                            /* Copy fapl, fcpl, and filepath name to new object */
```

Comment:
The reason for this change is that using the same logic for files and groups led to a memory error with `filepath_name`. 

Specifically, the group logic increments the reference count to the `hid_t` resources inside the parent object, such that the next call to close the resource only decreases the reference count. But because a file's `filepath_name` is a raw string, its reference count can't be increased. Thus, the previous logic would double free it.